extensions/msal-node-extensions/src/broker/NativeBrokerPlugin.ts

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccountInfo, AuthenticationResult, AuthenticationScheme, ClientAuthError, ClientConfigurationError, Constants, IdTokenClaims, INativeBrokerPlugin, InteractionRequiredAuthError, Logger, LoggerOptions, NativeRequest, NativeSignOutRequest, PromptValue, ServerError } from "@azure/msal-common";
+import { msalNodeRuntime, Account, AuthParameters, AuthResult, ErrorStatus, MsalRuntimeError, ReadAccountResult, DiscoverAccountsResult, SignOutResult, LogLevel as MsalRuntimeLogLevel} from "@azure/msal-node-runtime";
+import { ErrorCodes } from "../utils/Constants";
+import { NativeAuthError } from "../error/NativeAuthError";
+import { version, name } from "../packageMetadata";
+
+export class NativeBrokerPlugin implements INativeBrokerPlugin {
+    private logger: Logger;
+    isBrokerAvailable: boolean;
+    
+    constructor() {
+        const defaultLoggerOptions: LoggerOptions = {
+            loggerCallback: () => {},
+            piiLoggingEnabled: false
+        };
+        this.logger = new Logger(defaultLoggerOptions, name, version); // Default logger
+        this.isBrokerAvailable = msalNodeRuntime.StartupError ? false : true;
+    }
+
+    setLogger(loggerOptions: LoggerOptions): void {
+        this.logger = new Logger(loggerOptions, name, version);
+        const logCallback = (message: string, logLevel: MsalRuntimeLogLevel, containsPii: boolean) => {
+            switch(logLevel) {
+                case MsalRuntimeLogLevel.Trace:
+                    if (containsPii) {
+                        this.logger.tracePii(message);
+                    } else {
+                        this.logger.trace(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Debug: 
+                    if (containsPii) {
+                        this.logger.tracePii(message);
+                    } else {
+                        this.logger.trace(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Info:
+                    if (containsPii) {
+                        this.logger.infoPii(message);
+                    } else {
+                        this.logger.info(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Warning:
+                    if (containsPii) {
+                        this.logger.warningPii(message);
+                    } else {
+                        this.logger.warning(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Error:
+                    if (containsPii) {
+                        this.logger.errorPii(message);
+                    } else {
+                        this.logger.error(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Fatal:
```

Comment:
Fatal is new log type for JS libraries. Are we adding docs?

Reply:
This is mapping MsalRuntime error levels to our existing JS error levels. This is not adding anything new to our public surface.